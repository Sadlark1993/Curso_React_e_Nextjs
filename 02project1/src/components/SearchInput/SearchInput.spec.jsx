import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { SearchInput } from "."




describe('<SearchInput />', ()=>{
  it('should input value to be "testing..."', ()=>{
    const fn = jest.fn();
    render(<SearchInput handleIChange={fn} searchValue={'testing...'}/>);

    const screenConst = screen.getByPlaceholderText(/search/i);
    
    expect(screenConst).toBeInTheDocument();
    expect(screenConst.value).toBe('testing...');
  });

  it('should call handleChange function on each key pressed', ()=>{
    const fn = jest.fn();
    render(<SearchInput handleIChange={fn} searchValue={'a random value'}/>);

    const screenConst = screen.getByPlaceholderText(/search/i);
    const value = 'a random value';
    userEvent.type(screenConst, value);

    expect(screenConst.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);

  });

  it('should match snapshot', ()=>{
    const fn = jest.fn();
    render(<SearchInput handleIChange={fn} searchValue={'testing...'}/>);
    const screenConst = screen.getByPlaceholderText(/search/i);

    expect(screenConst).toMatchSnapshot();
  })
})