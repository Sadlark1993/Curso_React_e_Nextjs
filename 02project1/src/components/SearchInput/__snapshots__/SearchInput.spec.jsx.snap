// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<input
  placeholder="search"
  type="search"
  value="testing..."
/>
`;
